### PR TITLE
refactor(query): remove precision in timestamp type

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

remove precision in timestamp type

discussion:  https://github.com/datafuselabs/databend/discussions/8149

Fixes #issue

### CODE REVIEW

**Review by b41sh — APPROVED** [review]

**Review by drmingdrmer — COMMENTED** [review]
Everything is good except there should be a test case that loads the `TimestampType` message of the version `15`, similar to this:
https://github.com/datafuselabs/databend/blob/fb9c11c8c5cf5b7e9c18831dc66532896dd7d1b7/src/meta/proto-conv/tests/it/user_stage.rs#L305

File: src/query/ast/src/parser/expr.rs [hunk 1]
```diff
@@ -1126,9 +1126,7 @@ pub fn type_name(i: Input) -> IResult<TypeName> {
     let ty_date = value(TypeName::Date, rule! { DATE });
     let ty_datetime = map(
         rule! { (DATETIME | TIMESTAMP) ~ ( "(" ~ #literal_u64 ~ ")" )? },
```
Comment (RinChanNOWWW) [inline src/query/ast/src/parser/expr.rs:1128]: Also remove `#literal_u64` ?

Comment (sundy-li) [reply src/query/ast/src/parser/expr.rs:1128]: Some SQL dialects may need this, I prefer just keep it now.

**Review by drmingdrmer — APPROVED** [review]

**Review by RinChanNOWWW — APPROVED** [review]
LGTM

**Pull request merged**